report.md

```diff
@@ -203,7 +203,55 @@ spec:
+```
+openssl genpkey -algorithm RSA -pkeyopt rsa_keygen_bits:4096 -out keys/rsa-private-key.pem 
+```
+After filling a form on the web page and submitting it, the data will be sand through JavaScript server to main component which is Java server. There it will be saved in databases making it possible to ask the service about it from web whenever we want to, and propagated through PubSub to Python component and Go service where it will be encoded and send further (in our case the second Go component — decoder). We can observe events in our app using zipkin.
 ## 8. Demo deployment steps<a name="deployment"></a>
+
 ## 9. Summary – conclusions<a name="summary"></a>
+We managed to create a system with non-trivial business logic and quickly integrate components with one another using dapr, despite the fact that every service was written in different language and was worked on by many people. In terms of system integration, dapr proved to be a big help. It also provides a possibility to quickly switch middleware underneath communication method using only configuration files.
+
+On the other hand, due to dapr being a new technology, there are some places and functionalities that lack documentation and on top of that are implemented in non-standard way what leads to problems in using some components provided by dapr. The most noticeable case here is cryptography, which has literally no documentation, only case specific examples implemented in .NET and Go.
+
+During implementation of web application with dapr client, we also encountered some bugs related to JavaScript modules not cooperating with one another and having errors in library implementation. The easiest to show example here is dapr client of service invocation implemented in react app, which leads to errors in gRPC-client implementation that is used by dapr.
+
+To sum things up, there are parts of dapr that were implemented really well and prove to be useful in creating scalable and easily maintainable systems, but you need to be wary of parts that are still in alpha phase and lack documentation about them.
 ## 10. References<a name="references"></a>
+#### [Dapr documentation](https://docs.dapr.io/)
+- [Publish / Subscribe](https://docs.dapr.io/getting-started/quickstarts/pubsub-quickstart/)
+- [Service Invocation](https://docs.dapr.io/getting-started/quickstarts/serviceinvocation-quickstart/)
+- [State Management](https://docs.dapr.io/getting-started/quickstarts/statemanagement-quickstart/)
+- [Configuration](https://docs.dapr.io/getting-started/quickstarts/configuration-quickstart/)
```

Comment:
Missing secrets and observability. We do not have external configuration.

Reply:
observability does not have a website